### PR TITLE
chore(Messaggi a valore legale): [IAMVL-26] Persist attachments preferences for MVL

### PR DESCRIPTION
**Pull request by pietro909**
## Short description
Persist on the device the user's preferences for MVL. The only preference available to date is whether to show a warning for attachments or not.

## List of changes proposed in this pull request
- Update [DownloadAttachmentConfirmationBottomSheet](https://github.com/pagopa/io-app/pull/3681/files#diff-04d9d7c779eec6038ab9af0b57e77a474cfeda11a4052b29b668461fcecb04f3R35) to save the preference **once the user is done** and initialize them
- Update [MvlAttachments](https://github.com/pagopa/io-app/pull/3681/files#diff-2de5ec1ba6bf2598f6aa698a983c5aff6bc00042d3fba19c30ca411622beaffeR93) to sidestep the alert based on the preference
- Add tests for [MvlAttachments](https://github.com/pagopa/io-app/pull/3681/files#diff-4c3f5a56c849be98fa85cd976e88f1c5f275d8fe83d71bada54f4fd635abedc5R80) to ensure it honours the preferences
- Add [new reducer](ts/features/mvl/store/reducers/preferences.ts) for preferences and [tests](https://github.com/pagopa/io-app/pull/3681/files#diff-b496fc9a5a634def78118b0015b1388204dcefcf3b32a55c5e5bff49a3f098f5R5) 

### Dev-only updates
Added [two buttons](https://github.com/pagopa/io-app/pull/3681/files#diff-de3c1a64bcc8fdca4c8dbc5d62dba36c8c5ded27255052ecc6526e614b1e8f5bR413) under [ProfileMainScreen](ts/screens/profile/ProfileMainScreen.tsx) to **wipe the AsyncStorage** and **dump its content to console**.
The buttons are used in the video attached to test the new feature.

## How to test

The video (on Youtube due to its size) showcases the entire path:
1. fresh installation
2. open MVL
3. tap attachment
4. flag "don't ask again"
5. download attachment
6. close the app
7. open the app
8. open MVL
3. tap attachment
5. download attachment without warning

demo

### CODE REVIEW

File: ts/screens/profile/ProfileMainScreen.tsx [hunk 1]
```diff
@@ -409,6 +410,39 @@ class ProfileMainScreen extends React.PureComponent<Props, State> {
...
+                () => {
+                  /* eslint-disable no-console */
+                  console.log("[DUMP START]");
+                  AsyncStorage.getAllKeys()
+                    .then(keys => {
+                      console.log(`\tAvailable keys: ${keys.join(", ")}`);
+                      return Promise.all(
+                        keys.map(key =>
+                          AsyncStorage.getItem(key).then(value => {
+                            // if (/(features)|(euCovidCert)|(mvl)/.test(key)) {
```
Comment (Undermaken) [inline ts/screens/profile/ProfileMainScreen.tsx:433]: could we remove this commented code?

File: ts/screens/profile/ProfileMainScreen.tsx [hunk 2]
```diff
@@ -409,6 +410,39 @@ class ProfileMainScreen extends React.PureComponent<Props, State> {
...
+                            console.log(`\tValue for ${key}\n\t\t`, value);
+                            // }
+                          })
+                        )
+                      );
+                    })
+                    .then(() => console.log("[DUMP END]"))
```
Comment (Undermaken) [inline ts/screens/profile/ProfileMainScreen.tsx:440]:
```suggestion
                    .then(() => {
                      console.log("[DUMP END]");
                      showToast(
                        "Dump completed. Check logs in your console!",
                        "success"
                      );
                    }
```
Same of previous. It's just 💅🏻 
Skip if you don't like it

**Review by Undermaken — CHANGES_REQUESTED** [review]

**Review by fabriziofff — CHANGES_REQUESTED** [review]
Could you add a test for the new persisted store section in https://github.com/pagopa/io-app/blob/master/ts/boot/__tests__/persistedStore.test.ts#L5? 
In this way we can prevent future modification and missing migration of this store section.

File: ts/screens/profile/ProfileMainScreen.tsx (hunk 2 above)
Comment (pietro909) [reply ts/screens/profile/ProfileMainScreen.tsx:440]: I see why you are proposing it, though these buttons are here as a hideous reminder that _we didn't have better choices_.
I don't want to encourage this behavior if possible 😃

File: ts/screens/profile/ProfileMainScreen.tsx (hunk 1 above)
Comment (pietro909) [reply ts/screens/profile/ProfileMainScreen.tsx]: Good catch! Fixed [here](https://github.com/pagopa/io-app/pull/3681/commits/186aeb787ba33b385cc380ea2d76af770bf617fd)

File: ts/features/mvl/store/actions/index.ts [hunk 3]
```diff
@@ -15,4 +19,10 @@ export const mvlDetailsLoad = createAsyncAction(
   "MVL_DETAILS_FAILURE"
 )<UIMessageId, Mvl, WithUIMessageId<NetworkError>>();
 
-export type MvlActions = ActionType<typeof mvlDetailsLoad>;
+export const mvlPreferencesSetWarningForAttachments = createStandardAction(
```
[the remark this answers is not in the document]
Comment (pietro909) [reply ts/features/mvl/store/actions/index.ts:26]: Sure! [Here](https://github.com/pagopa/io-app/pull/3681/commits/186aeb787ba33b385cc380ea2d76af770bf617fd) it is

**Comment by pietro909** [comment]
@fabriziofff test added [here](https://github.com/pagopa/io-app/pull/3681/commits/cf8102538209c67121c3ecc96a68d6a2a2a07b2e). Thanks for pointing it out!

**Review by Undermaken — APPROVED** [review]

**Comment by Undermaken** [comment]
@fabriziofff do you want also review this PR?
For me, it looks good!

**Review by fabriziofff — APPROVED** [review]
I didn't review the whole pull request, just checked the persisted store test.

**Pull request merged**